### PR TITLE
WIP: Initial implementation of `safe_call` 

### PR DESCRIPTION
**Pull request by shaded-enmity**
There's a lot of TODOs throughout the code, this is really just a proof of concept.

- [ ] Use libc functions
- [ ] Drain both pipes via select/poll
- [ ] Use mount namespace

### CODE REVIEW

File: leapp/libraries/common/safe_call.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+        :param buffer_size: Size of the read buffer
+        :type buffer_size: int
+    '''
+    # first close the write end since we're just going to read
+    os.close(w)
+    b = os.read(r, buffer_size)
+    output = b
+    while b:
+        b = os.read(r, buffer_size)
+    output += b
```
Comment (jmikovic) [inline leapp/libraries/common/safe_call.py]: output += os.read() in loop

**Comment by vojtechsokol** [comment]
Could you please merge latest changes from master to be able to use the *copr_build* target from the Makefile?

**Comment by shaded-enmity** [comment]
@vojtechsokol Yeah though I first want to get this implementation on par with the `_call` implementation from the other PR

File: leapp/libraries/common/safe_call.py (hunk 1 above)
Comment (vinzenz) [inline leapp/libraries/common/safe_call.py]:
```suggestion
        output += b
```

**Comment by vinzenz** [comment]
https://github.com/oamg/leapp/blob/f6275a937a9ad5b18fcc033acde2088adbd5f631/leapp/libraries/common/safe_call.py#L110-L112

```python
    nothrow = kwargs.pop('nothrow')
```

**Comment by fernflower** [comment]
Do we still need this? I'd rather close it and reopen if necessary. @oamg/developers

**Comment by pirat89** [comment]
@fernflower I agree. currently we have bigger fish to catch and it's here already 3+ years. Probably better to close it and reopen when it's back on the table.

**Pull request closed without merging**